### DISCUSSION TITLE
A question about test coverage files

### DISCUSSION THREAD

**Discussion by flexbox** [nrwl/nx#11109] [Polls]
Hello,

One of my co-workers ran 

```
nx test web-ui --codeCoverage --html
nx test mobile --codeCoverage --html
```
and committed all the HTML reports files from jest to our repo.

I have the feeling adding all these files to the repo adds extra noise for nothing.

I have fixed the problem by deleting/ignoring 
```diff
# .gitignore

--coverage/
++**/coverage/**
```

Do you think it will create trouble in the future? and my general question is:

**Comment by rfprod** [comment]
@flexbox I think `no`, because when someone changes the structure of a monorepository, such as removing a library or replacing a library, unnecessary artifacts that no longer make sense will be present in the SCM unless they are removed manually. Storing coverage reports in SCM seems unmanageable at scale, it's a build artifact. I hope your colleague doesn't commit content of the `./dist` folder to SCM. If your team needs to keep the `./coverage` directory, you can use https://github.com/actions/upload-artifact.

**Reply by flexbox** [reply]
Makes sense to me :)

I think this should be updated then 
https://github.com/nrwl/nx/blob/master/packages/workspace/src/generators/workspace/files/__dot__gitignore#L30

**Reply by rfprod** [reply]
@flexbox  https://github.com/nrwl/nx/blob/master/packages/workspace/src/generators/workspace/files/__dot__gitignore#L30 - this `.gitignore` template looks valid to me. All coverage reports usually go to the `./coverage` directory in the project root. The coverage dist directory is declared in the library's jest configuration as shown below.
```typescript
const jestPresetAngularSerializers = require('jest-preset-angular/build/serializers');

const config = {
  preset: '../../jest.preset.js',
  coverageDirectory: '../../coverage/libs/client-pwa-offline',  // <-- coverage dist
  setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
  snapshotSerializers: jestPresetAngularSerializers,
  displayName: 'client-pwa-offline',
  coverageThreshold: {
    global: {
      branches: 100,
      functions: 100,
      lines: 100,
      statements: 100,
    },
  },
};

export default config;
```

**Reply by flexbox** [reply]
@rfprod Thanks for the input!

After some investigation, I found that my `/libs/web-ui/jest.config.ts` was missing `coverageDirectory`

Coverages files are exported to `/libs/web-ui/coverage/*`

<img width="271" alt="image" src="https://user-images.githubusercontent.com/360936/179187620-0f6831ec-c148-432d-ad47-e981585633d9.png">

Not really sure what happened when we created libraries 😅
